controllers/scripts/ipfs.go

```diff
@@ -0,0 +1,259 @@
+func applyFlatfsServer(conf *config.Config) error {
+	for _, profile := range []string{"flatfs", "server"} {
+		transformer, ok := config.Profiles[profile]
+		if !ok {
+			return fmt.Errorf("invalid configuration profile: %s", profile)
+		}
+
+		if err := transformer.Transform(conf); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// setFlatfsShardFunc Attempts to update the given flatfs configuration to use a shardFunc
+// with the given `n`. If unsuccessful, false will be returned.
+func setFlatfsShardFunc(conf *config.Config, n uint8) error {
+	// we want to use next-to-last/3 as the sharding function
+	// as per this issue:
+	// https://github.com/redhat-et/ipfs-operator/issues/32
+	mounts, ok := conf.Datastore.Spec["mounts"].([]interface{})
+	if !ok {
+		return fmt.Errorf("could not convert mounts to interface")
+	}
+
+	// set /repo/flatfs/shard/v1/next-to-last/3 as the shardFunc for flatfs
+	for _, mount := range mounts {
+		var mountMap, child map[string]interface{}
+		var dsType string
+		mountMap, ok = mount.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("could not convert mount to m: string -> interface")
+		}
+		// get child
+		child, ok = mountMap["child"].(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("could not convert child to map string to interface")
+		}
+		// get type
+		dsType, ok = child["type"].(string)
+		if !ok {
+			continue
+		}
+		if dsType == "flatfs" {
+			child["shardFunc"] = "/repo/flatfs/shard/v1/next-to-last/" + strconv.FormatUint(uint64(n), 10)
+			break
+		}
+	}
+	return nil
+}
+
+// applyIPFSClusterK8sDefaults Applies settings to the given Kubo configuration
+// which are customized specifically for running within a Kubernetes cluster.
+func applyIPFSClusterK8sDefaults(conf *config.Config, storageMax string, peers []peer.AddrInfo, rc config.RelayClient) {
+	conf.Addresses.API = config.Strings{"/ip4/0.0.0.0/tcp/5001"}
+	conf.Addresses.Gateway = config.Strings{"/ip4/0.0.0.0/tcp/8080"}
+	conf.Swarm.ConnMgr.HighWater = 2000
+	conf.Datastore.BloomFilterSize = 1048576
+	conf.Datastore.StorageMax = storageMax
+	conf.Swarm.EnableHolePunching = config.True
+
+	conf.Swarm.RelayClient = rc
+	conf.Peering.Peers = peers
+}
+
+// createTemplateConfig Returns a kubo configuration which contains preconfigured
+// settings that are optimized for running this within a Kubernetes cluster.
+func createTemplateConfig(
+	storageMax string,
+	peers []peer.AddrInfo,
+	rc config.RelayClient,
+) (conf config.Config, err error) {
+	// attempt to generate an identity
+	var identity config.Identity
+
+	// try to generate an elliptic curve key first
+	identity, err = config.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
+		options.Key.Type(options.Ed25519Key),
+	})
+	// fall back to RSA
+	if err != nil {
+		identity, err = config.CreateIdentity(os.Stdout, []options.KeyGenerateOption{
+			options.Key.Type(options.RSAKey),
```

Comment:
Can we use ECDSA. They are the default these days and look better in the config.

Reply:
We've dealt with clusters in the past which don't support key-generation through ECDSA so I've included RSA here as a fall-back option.
Should we error out completely if ECDSA is not available?

Reply:
Yes, I think this is not related and ECSDA should just work fine.